### ISSUE TITLE
Ignition-button should be a button element, for accessibility

### ISSUE THREAD

**Issue by boyum** [GetmeUK/ContentTools#335]
By using native `<button>` instead of making one out of a `<div>` for ignition buttons, quite a few functions are gained for free. `<button>` have click listeners for mouse, spacebar and return-key, and are natively tabbable. Screen readers interact way better with `<button>` than with `<div>`, and it's very easy to style them equally.

Of course, one can use aria `role="button"` and add click listeners with JavaScript, but why would you, when the browser does it all for you?

(Granted, I have not tried switching the divs for buttons, but I have never ran into problems previously)

Also, I might check into this when I have time for it. How often do you release updates of your application on npm and the like?

**Comment by boyum** [comment]
I made it work with `<button>`. Will file a pull request later today. Might not be of your standards, as I'm no CoffeeScript wiz, so please fix my code if it's needed.

**Comment by anthonyjb** [comment]
Hi @boyum - sorry for the later than intended reply,

I'm afraid I'm not convinced this change (at least on it's own) is justified. I understand that a button is better for screen readers but CT isn't designed for use in a screen reader - I'm struggling to see what we get from the change, we haven't removed the requirement to capture click events further this adds additional code to the stack for something that provides no additional functionality as far as I can see. 

The UI is designed for use with a mouse, and moving forward hopefully we'll look to add touch support to - but making the ignition buttons part of the pages tab flow isn't something I've considered and making buttons in the UI tab indexable and selectable with the return key in general seems like it would cause more issues that it would solve given the use of tab within the editable content itself. 

The use of `div` tags for every UI component (almost, they're not used for `input`s or `textarea`s for example) is deliberate, because I know the semantics of the ContentTools UI are for the most part unimportant and therefore I prefer to only think about structure in terms of CSS classes not HTML element types.

So I can't accept the pull request at this time I'm afraid. I don't want to put you of contributing, the code you submitted was in line with the existing library and had I agreed with the change I would have been happy to accept it.

Ant
